pkg/handler/repositories.go

```diff
@@ -198,9 +198,13 @@ func (rh *RepositoryHandler) update(c echo.Context, fillDefaults bool) error {
 	if err := c.Bind(&repoParams); err != nil {
 		return echo.NewHTTPError(http.StatusBadRequest, "error binding params: "+err.Error())
 	}
+
 	if fillDefaults {
 		repoParams.FillDefaults()
+	} else {
+		repoParams.FillNonNilValues()
```

Comment:
Hmm. So I suggested putting this at part of the beforecreate/beforeupdate hooks because then this would happen closer to the database level. 

With this way (the handler level), if I'm in the dao layer (somehow - maybe in a test) I could still create `versions = []`. For practical use cases I guess it's not important, but what do you think?

Reply:
Will try this.